x-pack/plugins/fleet/storybook/context/index.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import type { StoryContext } from '@storybook/react';
+import { createBrowserHistory } from 'history';
+
+import { I18nProvider } from '@kbn/i18n/react';
+
+import { ScopedHistory } from '../../../../../src/core/public';
+import { IntegrationsAppContext } from '../../public/applications/integrations/app';
+import type { FleetConfigType, FleetStartServices } from '../../public/plugin';
+
+// TODO: This is a contract leak, and should be on the context, rather than a setter.
+import { setHttpClient } from '../../public/hooks/use_request';
+
+import { getApplication } from './application';
+import { getChrome } from './chrome';
+import { getHttp } from './http';
+import { getUiSettings } from './ui_settings';
+import { getNotifications } from './notifications';
+import { stubbedStartServices } from './stubs';
+
+// TODO: clintandrewhall - this is not ideal, or complete.  The root context of Fleet applications
+// requires full start contracts of its dependencies.  As a result, we have to mock all of those contracts
+// with Storybook equivalents.  This is a temporary solution, and should be replaced with a more complete
+// mock later, (or, ideally, Fleet starts to use a service abstraction).
```

Comment:
Are there examples of this type of service abstraction in other Kibana applications that are using storybook? Would love to reference and give the Fleet team some context.

Reply:
Sure!  One implementation is the Plugin Services Toolkit from @elastic/kibana-presentation:

https://docs.elastic.dev/kibana-dev-docs/presentationPlugin
https://github.com/elastic/kibana/tree/master/src/plugins/presentation_util/public/services

We use this in Canvas to abstract consumed services.  For example:

- We define a service interface: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/services/platform.ts
- Kibana implementation: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/services/kibana/platform.ts
- Storybook and stub implementation: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/services/stubs/platform.ts
- Define hooks: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/services/index.ts
- Start services in Kibana: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/plugin.tsx#L121-L124
- Start services in Storybook: https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/storybook/decorators/services_decorator.tsx#L20-L24